src/nccl_ofi_rdma.c

```diff
@@ -6830,6 +6819,45 @@ static int ep_rail_init(nccl_net_ofi_rdma_ep_t *ep,
 	return 0;
 }
 
+static int get_domain_and_cq_for_endpoint_rail(nccl_net_ofi_rdma_ep_t *ep,
+					       nccl_net_ofi_rdma_device_rail_t *dev_rail,
+					       nccl_net_ofi_ep_rail_t *ep_rail,
+					       struct fid_domain **domain,
+					       struct fid_cq **cq)
+{
+	int ret = 0;
+
+	if (ep_rail->domain != NULL) {
+		*domain = ep_rail->domain;
+		*cq = ep_rail->cq;
+		return 0;
+	}
+	nccl_net_ofi_rdma_plugin_t *plugin = rdma_endpoint_get_plugin(ep);
+
+	if (plugin->base.domain_per_thread) {
+		assert(ep_rail->cq == NULL);
```

Comment:
Why is cq expected to be NULL here? If `ep_rail` is already initialized, shouldn't `ep_rail->cq` be non-NULL, even in domain-per-thread mode?